qlib/strategy/base.py

```diff
@@ -0,0 +1,179 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+import copy
+import pandas as pd
+from typing import List, Union
+
+
+from ..model.base import BaseModel
+from ..data.dataset import DatasetH
+from ..data.dataset.utils import convert_index_format
+from ..contrib.backtest.order import Order
+from ..rl.interpreter import ActionInterpreter, StateInterpreter
+from ..utils import init_instance_by_config
+
+
+class BaseStrategy:
+    """Base strategy for trading"""
+
+    def __init__(
+        self,
+        rely_trade_decision: object = None,
+        level_infra: dict = {},
+        common_infra: dict = {},
+    ):
+        """
+        Parameters
+        ----------
+        rely_trade_decision : object, optional
+            the high-level trade decison on which the startegy rely, and it will be traded in [start_time , end_time] , by default None
+            - If the strategy is used to split trade decison, it will be used
+            - If the strategy is used for portfolio management, it can be ignored
+        level_infra : dict, optional
+            level shared infrastructure for backtesting, including trade_calendar
+        common_infra : dict, optional
+            common infrastructure for backtesting, including trade_account, trade_exchange, .etc
+        """
+
+        self.reset(level_infra=level_infra, common_infra=common_infra, rely_trade_decision=rely_trade_decision)
+
+    def reset_level_infra(self, level_infra):
+        if not hasattr(self, "level_infra"):
+            self.level_infra = level_infra
+        else:
+            self.level_infra.update(level_infra)
+
+        if "trade_calendar" in level_infra:
+            self.trade_calendar = level_infra.get("trade_calendar")
+
+    def reset_common_infra(self, common_infra):
+        if not hasattr(self, "common_infra"):
+            self.common_infra = common_infra
+        else:
+            self.common_infra.update(common_infra)
+
+        if "trade_account" in common_infra:
+            self.trade_position = common_infra.get("trade_account").current
+
+    def reset(self, level_infra: dict = None, common_infra: dict = None, rely_trade_decision=None, **kwargs):
```

Comment:
What is the interface of:

- level_infra
- common_infra
- rely_trade_decision

When is reset expected to be called?